api/pipeline/pipeline_steps.py

```diff
@@ -59,26 +81,37 @@ async def formulate_response(sio: AgentStateManager, question: str) -> str:
         resp = await b.FormulateAnswer(question, context)
         if isinstance(resp, FinalAnswer):
             sio.add_action(type="ReadyToAnswer", content=resp.reason)
-            answer = await b.AnswerQuestion(question, context)
+            answer: str = await b.AnswerQuestion(question, context)
             sio.add_action(type="HumanApproval", content=answer)
-            res = await run_async(submit_answer_wrapper, question, answer)
-            if isinstance(res, tuple):
-                sio.add_action(type="Finalizing Answer", content=res[1])
-                return res[1]
+
+            approval_result = await run_approval(question, answer)
+
+            if approval_result.approved is True:
+                return answer
             else:
-                if "MY_EXIT_PROMPT" in res:
+                if "MY_EXIT_PROMPT" in approval_result.comment:
                     raise Exception("Human took over")
-                sio.add_action(type="Incorporating Feedback", content=res)
+                sio.add_action(
+                    type="Incorporating Feedback",
+                    content=f"Admin denied FinalAnswer with feedback: {approval_result.comment}",
+                )
```

Comment:
with the decorator, the comment got wrapped in "denied w/ feedback" message, but now we have to do that manually